components/playlist/browser/resources/components/playerSeeker.tsx

```diff
@@ -0,0 +1,253 @@
+      value: this.#progressElem.value
+    }
+
+    // Pause video during drag-and-drop session.
+    this.#videoRef.current.pause()
+
+    this.#videoRef.current.currentTime = this.interpolatePxToValue(
+      this.#initialState.clientX - this.#progressElem.offsetLeft
+    )
+
+    document.addEventListener('mousemove', this.onMouseMove)
+    document.addEventListener('mouseup', this.onMouseUp)
+  }
+
+  onMouseMove = (event: MouseEvent) => {
+    if (!this.#videoRef.current) {
+      return
+    }
+
+    if (!this.#isMouseDown) {
+      return
+    }
+
+    const dragThreshold = 5
+    if (
+      !this.#isDragging &&
+      Math.abs(event.clientX - this.#initialState!.clientX) > dragThreshold
+    ) {
+      this.#isDragging = true
+    }
+
+    if (!this.#isDragging) {
+      return
+    }
+
+    // Update value
+    const newTime = this.interpolatePxToValue(
+      event.clientX - this.#progressElem.offsetLeft
+    )
+
+    const oldTime = this.#videoRef.current.currentTime
+    this.#videoRef.current.currentTime = newTime
+
+    if (oldTime === this.#videoRef.current.currentTime) {
+      // In case the newTime has changed by time smaller than a second,
+      // the |currentTime| won't be updated. In this case <progress> could feel
+      // like jank. In order to avoid that, force to update the progress ui.
+      this.#updateProgressValue(newTime)
+    }
+  }
+
+  onMouseUp = (event: MouseEvent) => {
+    document.removeEventListener('mousemove', this.onMouseMove)
+    document.removeEventListener('mouseup', this.onMouseUp)
+
+    if (!this.#videoRef.current) {
+      return
+    }
+
+    this.#isMouseDown = false
+
+    if (this.#isDragging) {
+      this.#videoRef.current.currentTime = this.interpolatePxToValue(
+        event.clientX - this.#progressElem.offsetLeft
+      )
+    } else {
+      this.#videoRef.current.currentTime = this.interpolatePxToValue(
+        this.#initialState!.clientX - this.#progressElem.offsetLeft
+      )
+    }
+
+    // Restore the playing state.
+    if (this.#videoRef.current.paused && this.#initialState!.playing) {
+      this.#videoRef.current.play()
+    }
+  }
+}
+
+export default function PlayerSeeker ({ videoElementRef }: Props) {
+  const [duration, setDuration] = React.useState(0)
+  const [currentTime, setCurrentTime] = React.useState(0)
+
+  React.useEffect(() => {
+    const videoElem = videoElementRef.current
+    if (!videoElem) {
+      console.error(`Video element doesn't exist`)
+      return
+    }
+
+    const onTimeChanged = () => {
+      setCurrentTime(videoElem.currentTime)
+    }
+    const onLoaded = () => {
+      setDuration(videoElem.duration)
+      onTimeChanged()
+    }
+    videoElem.addEventListener('loadeddata', onLoaded)
+    videoElem.addEventListener('timeupdate', onTimeChanged)
+    return () => {
+      videoElem.removeEventListener('loadeddata', onLoaded)
+      videoElem.removeEventListener('timeupdate', onTimeChanged)
+    }
+  }, [videoElementRef.current])
```

Comment:
Similar story with `useEffect` here - better to pass the value of the ref through via state, as this won't necessarily be rerendered when the videoElement changes